service/history/stateBuilder.go

```diff
@@ -627,62 +673,83 @@ func (b *stateBuilderImpl) scheduleSignalWorkflowTransferTask(domainID string, w
 	}
 }
 
+func (b *stateBuilderImpl) scheduleUpsertSearchAttributesTask() persistence.Task {
+	return &persistence.UpsertWorkflowSearchAttributesTask{}
+}
+
 func (b *stateBuilderImpl) scheduleDeleteHistoryTransferTask() persistence.Task {
 	return &persistence.CloseExecutionTask{}
 }
 
-func (b *stateBuilderImpl) scheduleDecisionTimerTask(event *shared.HistoryEvent, scheduleID int64, attempt int64,
-	timeoutSecond int32) persistence.Task {
+// Timer tasks
+
+func (b *stateBuilderImpl) scheduleDecisionTimerTask(
+	event *shared.HistoryEvent,
+	scheduleID int64,
+	attempt int64,
+	timeoutSecond int32,
+) persistence.Task {
 	return b.getTimerBuilder(event).AddStartToCloseDecisionTimoutTask(scheduleID, attempt, timeoutSecond)
 }
 
-func (b *stateBuilderImpl) scheduleUserTimerTask(event *shared.HistoryEvent,
-	ti *persistence.TimerInfo, msBuilder mutableState) persistence.Task {
+func (b *stateBuilderImpl) scheduleUserTimerTask(
+	event *shared.HistoryEvent,
+	ti *persistence.TimerInfo,
+	msBuilder mutableState,
+) persistence.Task {
 	timerBuilder := b.getTimerBuilder(event)
 	timerBuilder.AddUserTimer(ti, msBuilder)
 	return timerBuilder.GetUserTimerTaskIfNeeded(msBuilder)
 }
 
-func (b *stateBuilderImpl) refreshUserTimerTask(event *shared.HistoryEvent, msBuilder mutableState) persistence.Task {
+func (b *stateBuilderImpl) refreshUserTimerTask(
+	event *shared.HistoryEvent,
+	msBuilder mutableState,
+) persistence.Task {
 	timerBuilder := b.getTimerBuilder(event)
 	timerBuilder.loadUserTimers(msBuilder)
 	return timerBuilder.GetUserTimerTaskIfNeeded(msBuilder)
 }
 
-func (b *stateBuilderImpl) scheduleActivityTimerTask(event *shared.HistoryEvent,
-	msBuilder mutableState) persistence.Task {
+func (b *stateBuilderImpl) scheduleActivityTimerTask(
+	event *shared.HistoryEvent,
+	msBuilder mutableState,
+) persistence.Task {
 	return b.getTimerBuilder(event).GetActivityTimerTaskIfNeeded(msBuilder)
 }
 
-func (b *stateBuilderImpl) scheduleWorkflowTimerTask(event *shared.HistoryEvent,
-	msBuilder mutableState) []persistence.Task {
+func (b *stateBuilderImpl) scheduleWorkflowTimerTask(
+	event *shared.HistoryEvent,
+	msBuilder mutableState,
+) []persistence.Task {
 	timerTasks := []persistence.Task{}
 	now := time.Unix(0, event.GetTimestamp())
-	timeout := now.Add(time.Duration(msBuilder.GetExecutionInfo().WorkflowTimeout) * time.Second)
-	backoffDuration := backoff.NoBackoff
-	startWorkflowAttribute := event.GetWorkflowExecutionStartedEventAttributes()
-	firstDecisionTaskBackoffSecond := startWorkflowAttribute.GetFirstDecisionTaskBackoffSeconds()
-	if firstDecisionTaskBackoffSecond > 0 {
-		backoffDuration = time.Duration(firstDecisionTaskBackoffSecond) * time.Second
-	}
 
-	if backoffDuration != backoff.NoBackoff {
-		timeout = timeout.Add(backoffDuration)
+	workflowTimeout := now.Add(time.Duration(msBuilder.GetExecutionInfo().WorkflowTimeout) * time.Second)
```

Comment:
Does this workflow timeout include the backoff duration?

Reply:
yes, plz see below, at line 733